modules/tinymce/src/plugins/directionality/test/ts/browser/DirectionalitySanityTest.ts

```diff
@@ -17,31 +17,272 @@ describe('browser.tinymce.plugins.directionality.DirectionalitySanityTest', () =
+                  dir: str.is('ltr')
+                },
+                children: [
+                  s.anything(),
+                  s.element('ul', {
+                    attrs: {
+                      dir: str.is('ltr')
+                    },
+                    children: [
+                      s.element('li', {}),
+                      s.element('li', {})
+                    ]
+                  }),
+                ]
+              })
+            ]
+          })
+        ]
+      });
+    }));
+
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Right to left"]');
+    TinyAssertions.assertContentStructure(editor, ApproxStructure.build((s, str) => {
+      return s.element('body', {
+        children: [
+          s.element('ul', {
+            attrs: {
+              dir: str.is('rtl')
+            },
+            children: [
+              s.element('li', {
+                attrs: {
+                  dir: str.is('ltr')
+                },
+                children: [
+                  s.anything(),
+                  s.element('ul', { // remove dir cause parent ul has rtl already
+                    children: [
+                      s.element('li', {}),
+                      s.element('li', {})
+                    ]
+                  }),
+                ]
+              })
+            ]
           })
         ]
       });
     }));
   });
 
-  it('TBA: Set and select content, click on the Left to right toolbar button and assert direction is left to right', () => {
+  it('TINY-4589: should remove dir attr if parent has same dir', () => {
     const editor = hook.editor();
-    editor.setContent('<p dir="rtl">a</p>');
+    editor.setContent(`
+    <div dir="ltr">
+    <p>foo</p>
+    <p>bar</p>
+    </div>
+    `);
     TinySelections.setSelection(editor, [ 0, 0 ], 0, [ 0, 0 ], 1);
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Right to left"]');
+    TinyAssertions.assertContent(editor, `<div dir="ltr">\n<p dir="rtl">foo</p>\n<p>bar</p>\n</div>`);
+
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Left to right"]');
+    TinyAssertions.assertContent(editor, `<div dir="ltr">\n<p>foo</p>\n<p>bar</p>\n</div>`);
+  });
+
+  it('TINY-4589: should not remove dir attr if parent has dir empty', () => {
+    const editor = hook.editor();
+    editor.setContent(`
+    <div dir="">
+      <p>foo</p>
+      <p>bar</p>
+    </div>
+    `);
+    TinySelections.setSelection(editor, [ 0, 0 ], 0, [ 0, 0 ], 1);
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Right to left"]');
+    TinyAssertions.assertContent(editor, `<div dir="">\n<p dir="rtl">foo</p>\n<p>bar</p>\n</div>`);
+
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Left to right"]');
+    TinyAssertions.assertContent(editor, `<div dir="">\n<p dir="ltr">foo</p>\n<p>bar</p>\n</div>`);
+  });
+
+  it('TINY-4589: should use first correct parent dir', () => {
+    const editor = hook.editor();
+    editor.setContent(`
```

Comment:
I'm pretty sure with `editor.setContent` this won't make a difference, but the extra newlines and whitespace in the HTML that you get from putting it all into a multiline template string like this can sometimes lead to extra text nodes, which throws off all of your `setSelection` API call numbers.

Shouldn't be an issue with `editor.setContent`, but keep it in mind if you ever need to use something like `SugarElement.fromHTML` in a test.

Reply:
Roight :)